frontend/src/metabase/components/DownloadButton.jsx

```diff
@@ -55,20 +72,53 @@ const handleSubmit = async (
 
   const formData = new URLSearchParams(new FormData(e.target));
 
-  const options = { method };
+  let options = { method };
   if (method === `POST`) {
     options.body = formData;
   } else if (method === `GET`) {
     options.query = formData.toString();
   }
 
-  fetch(method === `POST` ? url : url + "?" + options.query, options)
+  let requestUrl =
+    method === `POST` || checkApiIsFromDappLooker(url)
+      ? url
+      : url + "?" + options.query;
+
+  if (checkApiIsFromDappLooker(url)) {
+    requestUrl = requestUrl + `&coreUserId=${coreUserId}`;
+    options = { ...options };
+  }
+
+  fetch(requestUrl, options)
     .then(async res => {
-      const blob = await res.blob();
-      const url = URL.createObjectURL(blob);
+      const blobData = await res.blob();
+
+      // This try-catch is used only to catch newResponse.json() error when getting response as Blob data
+      try {
+        const newResponse = new Response(blobData);
+        const downloadResponse = (await newResponse?.json()) || undefined;
+        if (
+          downloadResponse?.success === false &&
+          checkApiIsFromDappLooker(requestUrl)
+        ) {
+          // const keys = Object.keys(downloadResponse.errorData);
+          // const errorMsg = downloadResponse.errorData[keys[0]];
+          // console.log(errorMsg);
+          onDownloadRejected();
+          return;
+        }
+      } catch (error) {
+        // console.error(error);
```

Comment:
Log this exception

Reply:
removed the log because this catch block should be empty